### PR TITLE
feat(sdk): network and entrypoint options for `sim.Container`

### PR DESCRIPTION
**Pull request by eladcon**
## Checklist

- [ ] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [ ] Description explains motivation and solution
- [ ] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

File: examples/tests/sdk_tests/container/my-docker-image/index.js [hunk 1]
```diff
@@ -6,9 +6,11 @@ process.on('SIGINT', () => {
   process.exit(0)
 });
 
+const response = process.env.RESPONSE;
+
 const server = http.createServer((req, res) => {
   console.log(`request received: ${req.method} ${req.url}`);
-  res.end('Hello, Wingnuts!');
+  res.end(response || 'Hello, Wingnuts!');
```
Comment (eladb) [inline examples/tests/sdk_tests/container/my-docker-image/index.js:13]:
```suggestion
  res.end(response ?? 'Hello, Wingnuts!');
```

File: examples/tests/sdk_tests/container/network.test.w [hunk 2]
```diff
@@ -0,0 +1,25 @@
+/*\
+skipPlatforms:
```
Comment (eladb) [inline examples/tests/sdk_tests/container/network.test.w:2]: why?

File: libs/wingsdk/src/target-sim/container.ts [hunk 3]
```diff
@@ -53,6 +53,18 @@ export interface ContainerProps {
    */
   readonly args?: string[];
 
+  /**
+   * Container network
```
Comment (eladb) [inline libs/wingsdk/src/target-sim/container.ts:57]: Any reference for possible values?

File: libs/wingsdk/src/target-sim/container.inflight.ts [hunk 4]
```diff
@@ -143,6 +152,15 @@ export class Container implements IContainerClient, ISimulatorResourceInstance {
 
       // if we are waiting for a port, check if the container is listening to it
       if (this.props.containerPort) {
+        // when using the host network, the host port is the same as the container port
```
Comment (eladb) [inline libs/wingsdk/src/target-sim/container.inflight.ts:155]: Test for this?

**Review by skyrpex — APPROVED** [review]
LGTM.

**Pull request merged**